src/commcare_cloud/commands/terraform/aws.py

```diff
@@ -28,45 +28,81 @@ def aws_cli(environment, cmd_parts):
+    ])
+
+    resources = {}
+    for name, ip in name_private_ip_pairs:
+        resources[name] = ip
+
+    for name, ip in name_public_ip_pairs:
+        resources['{}.public_ip'.format(name)] = ip
+
+    if elasticache_address:
+        resources['redis-{}'.format(config.environment)] = elasticache_address
+
+    return resources
+
+
 class AwsList(CommandBase):
     command = 'aws-list'
     help = "List endpoints (ec2, rds, etc.) on AWS"
 
     def run(self, args, unknown_args):
         environment = get_environment(args.env_name)
-        config = environment.terraform_config
-
-        # Private IP addresses
-        private_ip_query = aws_cli(environment, [
-            'aws', 'ec2', 'describe-instances',
-            '--filter', "Name=tag-key,Values=Name", "Name=tag-value,Values=*",
-            "Name=instance-state-name,Values=running",
-            "Name=tag-key,Values=Environment",
-            "Name=tag-value,Values={}".format(config.environment),
-            "--query", "Reservations[*].Instances[*][Tags[?Key=='Name'].Value, NetworkInterfaces[0].PrivateIpAddresses[0].PrivateIpAddress]",
-            "--output", "json",
-            "--region", config.region,
-        ])
-        name_private_ip_pairs = [(item[0][0][0], item[0][1]) for item in private_ip_query]
-
-        # Public IP addresses
-        public_ip_query = aws_cli(environment, [
-            'aws', 'ec2', 'describe-instances',
-            '--filter', "Name=tag-key,Values=Name", "Name=tag-value,Values=*",
-            "Name=instance-state-name,Values=running",
-            "Name=tag-key,Values=Environment",
-            "Name=tag-value,Values={}".format(config.environment),
-            "--query",
-            "Reservations[*].Instances[*][Tags[?Key=='Name'].Value[],NetworkInterfaces[0].Association.PublicIp]",
-            "--output", "json",
-            "--region", config.region,
-        ])
-        name_public_ip_pairs = [(item[0][0][0], item[0][1]) for item in public_ip_query
-                                if item[0][1] is not None]
-
-        for name, ip in name_private_ip_pairs:
-            print('{}\t{}'.format(ip, name))
-
-        for name, ip in name_public_ip_pairs:
-            print('{}\t{}.public_ip'.format(ip, name))
+        resources = get_aws_resources(environment)
+        for name, address in sorted(resources.items()):
+            print('{}\t{}'.format(address, name))
         return 0
+
+
+class AwsFillInventory(CommandBase):
+    command = 'aws-fill-inventory'
+    help = "Fill inventory.ini.j2 using existing AWS resources"
+
+    def run(self, args, unknown_args):
+        environment = get_environment(args.env_name)
+        resources = get_aws_resources(environment)
+        with open(environment.paths.inventory_ini_j2) as f:
+            template_string = f.read()
+
+        for name, address in resources.items():
+            template_string = template_string.replace('{{ ' + name + ' }}', address)
```

Comment:
So if this is to replace the parts of the inventory.ini.j2 file that are nested in curly braces, how come some of the groups are hardcoded ie https://github.com/dimagi/commcare-cloud/pull/2220/files#diff-b1724c5eb58f38c295df121214624b47R91 ?

Reply:
There are a lot of lines that are commented out in the staging inventory file right now, like the one you pointed out, that are essentially placeholders for machines that don't exist yet. (The IP addresses that are there are IP of machines that existed at some point when I was testing some things a couple months back, but have since been destroyed.)

Reply:
Sounds good! A comment to that effect could be useful